test/custom-header.spec.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+const expect = require('chai').expect;
+
+module.exports = (fetchMock, theGlobal, Headers) => {
+
+    describe('fetch-mock', () => {
+
+        const dummyFetch = () => Promise.resolve(arguments);
+
+        before(() => {
+            theGlobal.fetch = dummyFetch;
+        })
+
+        describe('Interface', () => {
+
+            it('handles Headers class configuration correctly', () => {
+                fetchMock.mock({
+                    name: 'route1',
+                    headers: {
+                        test: 'yes'
+                    },
+                    matcher: 'http://it.at.here/',
+                    response: 'ok'
+                }).catch();
+                return Promise.all([
+                    fetch('http://it.at.here/', {headers: {test: 'yes'}}),
+                    fetch('http://it.at.here/', {headers: new Headers({test: 'yes'})}),
+                    fetch('http://it.at.here/')
+                ])
+                    .then(() => {
+                        expect(fetchMock.called()).to.be.true;
+                        expect(fetchMock.called('route1')).to.be.true;
+                        expect(fetchMock.calls('route1').length).to.equal(2);
+                        expect(fetchMock.calls().matched.length).to.equal(2);
+                        expect(fetchMock.calls().unmatched.length).to.equal(1);
+                    });
+            });
+
+            it('handles headers with multiple values correctly', () => {
+                fetchMock.mock({
+                    name: 'route1',
+                    headers: {
+                        test: ['foo', 'bar']
+                    },
+                    matcher: 'http://it.at.here/',
+                    response: 'ok'
+                }).catch();
+                return Promise.all([
+                    fetch('http://it.at.here/', {headers: {test: 'yes'}}),
+                    fetch('http://it.at.here/', {headers: {test: ['foo', 'bar']}}),
```

Comment:
**Case 2**

Case 1 led me to a general question. Even if not using the mentioned `Headers` class, a custom HTTP header may have _n_ values (see section 4.2 of [RFC-2616](https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html)).

_Example API call_
```javascript
// With Headers class
const headers = new Headers();
headers.append('X-WEBMASTER', 'CHUCK NORRIS');
headers.append('X-WEBMASTER', 'BATMAN');

fetch('/api', {
    'method': 'POST',
    'headers': headers,
})

// With plain notation
fetch('/api', {
    'method': 'POST',
    'headers': {'X-WEBMASTER': ['CHUCK NORRIS', 'BATMAN']},
})

```

**Question**: Is there a way of testing a HTTP request with such header setup with `fetch-mock`? What's your suggestion on it?